common/core/ui_core_confirm.c

```diff
@@ -98,8 +98,18 @@
  * GLOBAL FUNCTIONS
  *****************************************************************************/
 
-bool core_user_confirmation(const char *msg, ui_core_rejection_cb *reject_cb) {
-  confirm_scr_init(msg);
+bool core_user_confirmation(const char *title,
+                            const char *body,
+                            ui_type_e type,
+                            ui_core_rejection_cb *reject_cb) {
+  if (SCROLL_PAGE_SCREEN == type) {
+    ui_scrollable_page(title, body, MENU_SCROLL_HORIZONTAL, false);
+  } else if (CONFIRMATION_SCREEN == type) {
+    confirm_scr_init(body);
+  } else {
+    // unexpected ui_type_e received; should never reach here
+    return false;
+  }
```

Comment:
We should split this change into two functions instead of adding more arguments.

The common part can be left common with a static function.

Reply:
changes added